### ISSUE TITLE
Invalid input to timestamp parameter breaks GUI

### ISSUE THREAD

**Issue by predictiple** [Velocidex/velociraptor#3922]
version: 0.73.3
commit: 6357f70ab
build_time: "2024-11-26T17:27:39+02:00"
compiler: go1.23.2
system: linux
architecture: amd64

Using
```yaml
parameters:
  - name: StartDate
    type: timestamp
    default: "now() - 2592000"
```
in a notebook template or artifact breaks the gui.

If the parameter is in a notebook template then when creating a notebook from the template, as soon as you select the template it tries to render the preview and you get a blank page. The same happens if such a paramter value is used in a server artifact. So it seems to be that the timestamp type just needs better handling of invalid values.

I've also confirmed that other invalid input such as an unparseable string produces the same result.

As a sidenote, it actually would be very useful if VQL such as `now() - 86400` was allowed in timestamp parameters. This would allow us to have notebooks and artifacts which timebox to an dynamic interval instead of fixed dates.

**Comment by scudette** [comment]
Thanks for reporting it - the JS is raising because the default value is not a valid timestamp (it must be an ISO time). We should probably add some checks to the validator to catch artifact bugs like this.

> As a sidenote, it actually would be very useful if VQL such as now() - 86400 was allowed in timestamp parameters. This would allow us to have notebooks and artifacts which timebox to an dynamic interval instead of fixed dates.

This can be implemented in VQL and should not be implemented in the parameter itself (if only for security reasons - any user can set a parameter - we dont want a situation where a parameter can be evaluated as code). As it is the GUI provides an easy way to set relative time to now.

**Comment by scudette** [comment]
This should be fixed in #3925
